packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkBlockchain.ts

```diff
@@ -247,16 +247,25 @@ export class ForkBlockchain
       rpcBlock === null ||
       rpcBlock.hash === null ||
       rpcBlock.number === null ||
-      rpcBlock.number.gt(this._forkBlockNumber)
+      rpcBlock.number > this._forkBlockNumber
     ) {
       return undefined;
     }
 
+    const forkNetworkId = this._jsonRpcClient.getNetworkId();
```

Comment:
This is cached internally, right?

Reply:
also, it should be chainId, right?

Reply:
> This is cached internally, right?

It's just a getter for a private field of the client. Maybe making that field `public readonly` would make this clearer, wdyt?

> also, it should be chainId, right?

I'm not sure.It's the value returned by the `getNetworkId` function, which uses `net_version`. Maybe that's a mistake.